data/provider/src/main/java/tech/pegasys/teku/api/ChainDataProvider.java

```diff
@@ -266,11 +257,13 @@ public boolean isFinalized(final UInt64 slot) {
     }
 
     if (validatorParameter.toLowerCase().startsWith("0x")) {
+      final Bytes48 keyBytes = getBytes48FromParameter(validatorParameter);
       try {
-        BLSPubKey publicKey = BLSPubKey.fromHexString(validatorParameter);
-        return spec.getValidatorIndex(state, publicKey.asBLSPublicKey());
-      } catch (PublicKeyException ex) {
-        throw new BadRequestException(String.format("Invalid public key: %s", validatorParameter));
+        // validate key against g2 curve
+        return spec.getValidatorIndex(state, BLSPublicKey.fromBytesCompressedValidate(keyBytes));
```

Comment:
I thought the aim was to *not* validate against the g2 curve?  I'd have just called `fromBytesCompressed` here (and then we shouldn't need the try/catch as we wouldn't be doing the G2 validation.